Hover/UI/HoverView.swift

```diff
@@ -134,7 +134,7 @@ public extension HoverView {
         let hitView = super.hitTest(point, with: event)
         if hitView == self {
             onTouchInDim()
-            return nil
```

Comment:
This was intentional back then 🤔

Reply:
If the DimView is visible and I tap in a button behind it, it's triggering the button action. With the fix, it will dismiss the HoverView without triggering any action behind the DimView.

Reply:
Yeah, I'm aware, what I meant is: that behaviour was intentional back then as a way to make it feel faster, I'm wondering if removing it is the best option 🤔

Reply:
Oh, alright. Well, I think the fix brings a better user experience, otherwise, the user can see two transitions at the same time, the DimView dismissal + a push or modal transition.

Reply:
> Well, I think the fix brings a better user experience, otherwise, the user can see two transitions at the same time, the DimView dismissal + a push or modal transition.

Yeah, you're probably right - as a framework that should be the default behaviour.